pyracf/common/utilities/logger.py

```diff
@@ -191,7 +258,11 @@ def redact_request_xml(
             # <tag operation="del" />
             if f"</{xml_key}>" not in xml_string:
                 continue
-            xml_string = self.__redact_string(xml_string, match.end(), f"</{xml_key}")
+            xml_string = re.sub(
+                rf"{xml_key}(.*)>.*<\/{xml_key}",
+                rf"{xml_key}\1>********</{xml_key}",
+                xml_string,
+            )
```

Comment:
Initialize regexes at the beginning for clarity and to increase signal to noise ratio?

```python
is_bytes = False
? = rf"\<{xml_key}+[^>]*\>"
start_tag_end_tag_regex = rf"{xml_key}(.*)>.*<\/{xml_key}"
redacted_regex = rf"{xml_key}\1>********</{xml_key}"
```

Reply:
Yes, the initial tag regex is also no longer necessary. Since these use the xml_key from this for loop, the definitions could not go all the way back to where is_bytes is defined.